### PR TITLE
Unit tests for binlog

### LINKED ISSUE #291: Add unit testing for package `binlog`
Add unit testing for `internal/stackql/acid/binlog`.

Acceptance Criteria:

- 85% unit test coverage.

### PR DESCRIPTION
**Pull request by BulkBeing**
## Type of change

- [ ] Bug fix (non-breaking change to fix a bug).
- [ ] Feature (non-breaking change to add functionality).
- [ ] Breaking change.
- [x] Other (eg: documentation change).  **Please explain**.

## Issues referenced.
Closes #291 

## Evidence

```
$ go test -cover -tags json1,sqleanall  ./...
?       github.com/stackql/stackql/internal/stackql/acid/acid_dto       [no test files]
ok      github.com/stackql/stackql/internal/stackql/acid/binlog 0.002s  coverage: 100.0% of statements
?       github.com/stackql/stackql/internal/stackql/acid/operation      [no test files]
?       github.com/stackql/stackql/internal/stackql/acid/transact       [no test files]
?       github.com/stackql/stackql/internal/stackql/acid/txn_context    [no test files]
?       github.com/stackql/stackql/internal/stackql/astanalysis/annotatedast    [no test files]
?       github.com/stackql/stackql/internal/stackql/astanalysis/earlyanalysis   [no test files]
?       github.com/stackql/stackql/internal/stackql/astanalysis/routeanalysis   [no test files]
?       github.com/stackql/stackql/internal/stackql/astanalysis/selectmetadata  [no test files]
?       github.com/stackql/stackql/internal/stackql/astformat   [no test files]
?       github.com/stackql/stackql/internal/stackql/astfuncrewrite      [no test files]
?       github.com/stackql/stackql/internal/stackql/astindirect [no test files]
?       github.com/stackql/stackql/internal/stackql/astvisit    [no test files]
?       github.com/stackql/stackql/internal/stackql/asyncmonitor        [no test files]
?       github.com/stackql/stackql/internal/stackql/bundle      [no test files]
?       github.com/stackql/stackql/internal/stackql/cmd [no test files]
?       github.com/stackql/stackql/internal/stackql/color       [no test files]
?       github.com/stackql/stackql/internal/stackql/config      [no test files]
?       github.com/stackql/stackql/internal/stackql/constants   [no test files]
?       github.com/stackql/stackql/internal/stackql/data_staging/input_data_staging     [no test files]
?       github.com/stackql/stackql/internal/stackql/data_staging/output_data_staging    [no test files]
?       github.com/stackql/stackql/internal/stackql/dataflow    [no test files]
?       github.com/stackql/stackql/internal/stackql/datasource/sql_datasource   [no test files]
?       github.com/stackql/stackql/internal/stackql/db_util     [no test files]
?       github.com/stackql/stackql/internal/stackql/dbmsinternal        [no test files]
?       github.com/stackql/stackql/internal/stackql/dependencyplanner   [no test files]
?       github.com/stackql/stackql/internal/stackql/discovery   [no test files]
?       github.com/stackql/stackql/internal/stackql/docparser   [no test files]
ok      github.com/stackql/stackql/internal/stackql/datasource/sqltable 0.008s  coverage: 100.0% of statements
?       github.com/stackql/stackql/internal/stackql/drm [no test files]
?       github.com/stackql/stackql/internal/stackql/dto [no test files]
?       github.com/stackql/stackql/internal/stackql/entryutil   [no test files]
?       github.com/stackql/stackql/internal/stackql/garbagecollector    [no test files]
?       github.com/stackql/stackql/internal/stackql/gcexec      [no test files]
?       github.com/stackql/stackql/internal/stackql/google_sdk  [no test files]
?       github.com/stackql/stackql/internal/stackql/handler     [no test files]
?       github.com/stackql/stackql/internal/stackql/httpmiddleware      [no test files]
?       github.com/stackql/stackql/internal/stackql/internal_data_transfer/builder_input        [no test files]
?       github.com/stackql/stackql/internal/stackql/internal_data_transfer/internaldto  [no test files]
?       github.com/stackql/stackql/internal/stackql/internal_data_transfer/primitive_context    [no test files]
?       github.com/stackql/stackql/internal/stackql/internal_data_transfer/relationaldto        [no test files]
?       github.com/stackql/stackql/internal/stackql/kstore      [no test files]
?       github.com/stackql/stackql/internal/stackql/netutils    [no test files]
?       github.com/stackql/stackql/internal/stackql/output      [no test files]
?       github.com/stackql/stackql/internal/stackql/parser      [no test files]
?       github.com/stackql/stackql/internal/stackql/parserutil  [no test files]
?       github.com/stackql/stackql/internal/stackql/plan        [no test files]
?       github.com/stackql/stackql/internal/stackql/planbuilder [no test files]
?       github.com/stackql/stackql/internal/stackql/metadatavisitors    [no test files]
?       github.com/stackql/stackql/internal/stackql/primitive   [no test files]
?       github.com/stackql/stackql/internal/stackql/primitivebuilder    [no test files]
?       github.com/stackql/stackql/internal/stackql/primitivecomposer   [no test files]
?       github.com/stackql/stackql/internal/stackql/primitivegenerator  [no test files]
?       github.com/stackql/stackql/internal/stackql/primitivegraph      [no test files]
?       github.com/stackql/stackql/internal/stackql/provider    [no test files]
?       github.com/stackql/stackql/internal/stackql/methodselect        [no test files]
?       github.com/stackql/stackql/internal/stackql/iqlerror    [no test files]
?       github.com/stackql/stackql/internal/stackql/logging     [no test files]
?       github.com/stackql/stackql/internal/stackql/planbuilderinput    [no test files]
?       github.com/stackql/stackql/internal/stackql/iqlutil     [no test files]
?       github.com/stackql/stackql/internal/stackql/nativedb    [no test files]
?       github.com/stackql/stackql/internal/stackql/providerconfig      [no test files]
?       github.com/stackql/stackql/internal/stackql/relational  [no test files]
?       github.com/stackql/stackql/internal/stackql/sql_system  [no test files]
?       github.com/stackql/stackql/internal/stackql/sqlcontrol  [no test files]
?       github.com/stackql/stackql/internal/stackql/sqlengine   [no test files]
?       github.com/stackql/stackql/internal/stackql/requests    [no test files]
?       github.com/stackql/stackql/internal/stackql/sqlrewrite  [no test files]
?       github.com/stackql/stackql/internal/stackql/sqlstream   [no test files]
?       github.com/stackql/stackql/internal/stackql/streaming   [no test files]
?       github.com/stackql/stackql/internal/stackql/streaming/http_preparator_stream.go [no test files]
?       github.com/stackql/stackql/internal/stackql/suffix      [no test files]
?       github.com/stackql/stackql/internal/stackql/symtab      [no test files]
?       github.com/stackql/stackql/internal/stackql/tableinsertioncontainer     [no test files]
?       github.com/stackql/stackql/internal/stackql/tablemetadata       [no test files]
?       github.com/stackql/stackql/internal/stackql/tablenamespace      [no test files]
?       github.com/stackql/stackql/internal/stackql/taxonomy    [no test files]
?       github.com/stackql/stackql/internal/stackql/templatenamespace   [no test files]
?       github.com/stackql/stackql/internal/stackql/typing      [no test files]
?       github.com/stackql/stackql/internal/stackql/util        [no test files]
?       github.com/stackql/stackql/internal/stackql/writer      [no test files]
?       github.com/stackql/stackql/internal/test/stackqltestutil        [no test files]
?       github.com/stackql/stackql/internal/test/testhttpapi    [no test files]
?       github.com/stackql/stackql/internal/test/testobjects    [no test files]
?       github.com/stackql/stackql/internal/test/testutil       [no test files]
?       github.com/stackql/stackql/pkg/awssign  [no test files]
?       github.com/stackql/stackql/pkg/azureauth        [no test files]
?       github.com/stackql/stackql/pkg/preprocessor     [no test files]
?       github.com/stackql/stackql/pkg/prettyprint      [no test files]
?       github.com/stackql/stackql/pkg/sqltypeutil      [no test files]
?       github.com/stackql/stackql/pkg/textutil [no test files]
?       github.com/stackql/stackql/pkg/txncounter       [no test files]
?       github.com/stackql/stackql/internal/stackql/responsehandler     [no test files]
?       github.com/stackql/stackql/internal/stackql/router      [no test files]
?       github.com/stackql/stackql/internal/stackql/session     [no test files]
?       github.com/stackql/stackql/internal/stackql/router/obtain_context       [no test files]
?       github.com/stackql/stackql/internal/stackql/sqlmachinery        [no test files]
ok      github.com/stackql/stackql/internal/stackql/driver      40.725s coverage: 24.6% of statements
ok      github.com/stackql/stackql/internal/stackql/psqlwire    0.003s  coverage: 0.0% of statements
ok      github.com/stackql/stackql/internal/stackql/querysubmit 0.658s  coverage: 29.4% of statements
ok      github.com/stackql/stackql/stackql      31.020s coverage: 50.0% of statements
```

## Checklist:

- [ ] A full round of testing has been completed, and there are no test failures as a result of these changes.
- [ ] The changes are covered with functional and/or integration robot testing.
- [ ] The changes work on all supported platforms.

### Variations

None

## Tech Debt

None

### CODE REVIEW

File: internal/stackql/acid/binlog/log.go [hunk 1]
```diff
@@ -52,10 +61,8 @@ func (l *simpleLogEntry) Size() int {
 func (l *simpleLogEntry) Clone() LogEntry {
 	rawCopy := make([]byte, len(l.raw))
 	copy(rawCopy, l.raw)
-	var humanReadableCopy []string
-	for _, s := range l.humanReadable {
-		humanReadableCopy = append(humanReadableCopy, strings.Clone(s))
```
Comment (BulkBeing) [inline internal/stackql/acid/binlog/log.go:57]: Strings are immutable anyway. This clone seems unnecessary.

File: internal/stackql/acid/binlog/log.go [hunk 2]
```diff
@@ -10,7 +8,7 @@ type LogEntry interface {
 	AppendHumanReadable(string)
 	AppendRaw([]byte)
 	Clone() LogEntry
-	Concatenate(LogEntry) LogEntry
```
Comment (BulkBeing) [inline internal/stackql/acid/binlog/log.go:13]: A separate `Clone` method is already provided. So it seems better to have the `Concatenate` method to mutate the `LogEntry` in-place.

**Comment by general-kroll-4-life** [comment]
@BulkBeing looks like linting check is failing, you can run lcally as per [the developer duide](https://github.com/stackql/stackql/blob/main/docs/developer_guide.md#linting).  Honestly, we need to make this coherent it can improve

**Comment by BulkBeing** [comment]
@general-kroll-4-life Made the changes to fix the linter errors.

**Review by general-kroll-4-life — APPROVED** [review]

**Comment by jeffreyaven** [comment]
great job @BulkBeing ! can you make sure to give us a ⭐ at https://github.com/stackql/stackql, thanks again!

**Pull request merged**